optimum/intel/openvino/modeling_decoder.py

```diff
@@ -436,30 +446,29 @@ def forward(
             inputs["beam_idx"] = self.next_beam_idx
```

Comment:
The most critical comment: `self.next_beam_idx` is a shared resource that is updated concurrently from multiple threads. It will lead to incorrect behavior when multiple `generate`s with different batch size or with beam search mode and different prompts, or with any kind of sampling are called. Separate version of `next_beam_idx` should be created for each generate.

Reply:
Indeed it is failing with parallel generate with different batch sizes. I'll try passing next_beam_idx also in kwargs along with infer_request.